packages/core/components/DataSourcePrompt/index.tsx

```diff
@@ -167,43 +171,28 @@ export default function DataSourcePrompt(props: Props) {
                 </div>
             )}
             <hr className={styles.divider} />
-            <div className={styles.actionsContainer}>
-                <form>
-                    <label
-                        aria-label="Browse for a data source file on your machine"
-                        title="Browse for a data source file on your machine"
-                        htmlFor="data-source-selector"
-                    >
-                        <PrimaryButton
-                            iconName="DocumentSearch"
-                            text="Choose file"
-                            title="Choose file"
-                        />
-                    </label>
-                    <input
-                        hidden
-                        accept=".csv,.json,.parquet"
-                        type="file"
-                        id="data-source-selector"
-                        name="data-source-selector"
-                        onChange={onChooseFile}
+            <FilePrompt onSelectFile={onSelectSource} selectedFile={dataSource} />
+            <div
+                className={classNames(styles.checkboxFormContainer, {
+                    [styles.selected]: hasMetadataSource,
+                })}
+            >
+                <div className={styles.checkboxContainer}>
+                    <Checkbox
+                        className={classNames(styles.checkbox, {
+                            [styles.selected]: hasMetadataSource,
+                        })}
+                        checked={hasMetadataSource}
+                        onChange={onToggleHasMetadataSource}
                     />
-                </form>
-                <div className={styles.orDivider}>OR</div>
-                <form className={styles.urlForm} onSubmit={onEnterURL}>
-                    <TextField
-                        onChange={(_, newValue) => setDataSourceURL(newValue || "")}
-                        placeholder="Paste URL (ex. S3, Azure)..."
-                        iconProps={{
-                            className: classNames(styles.submitIcon, {
-                                [styles.disabled]: !dataSourceURL,
-                            }),
-                            iconName: "ReturnKey",
-                            onClick: onEnterURL,
-                        }}
-                        value={dataSourceURL}
+                    <p onClick={onToggleHasMetadataSource}>Input additional metadata source</p>
```

Comment:
Nit: This prompt feels slightly confusing to me, especially since it's right next to the url field in a way that looks like they're paired together. I initially interpret this as just "add another source with more data in it" instead of a separate file with metadata describing the actual file. I'm not sure if there's a clearer way to put that though.
Visually, maybe if the dashed border were around "choose file OR paste url" instead so those are grouped as a single step with this outside?
<img width="400" alt="image" src="https://github.com/user-attachments/assets/4baedbb9-9943-442f-86d9-5ee2f8dc23bc">

Reply:
Styled according to conversation we had with UX team following this comment